File: backend/src/main/java/com/hospital/hms/service/LabReportService.java
package com.hospital.hms.service;

import com.hospital.hms.dto.request.LabReportRequest;
import com.hospital.hms.dto.request.LabReportUpdateRequest;
import com.hospital.hms.dto.response.LabReportResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface LabReportService {

    LabReportResponse request(LabReportRequest request);

    LabReportResponse update(Long id, LabReportUpdateRequest request);

    LabReportResponse getById(Long id);

    Page<LabReportResponse> getByPatient(Long patientId, Pageable pageable);

    Page<LabReportResponse> getByDoctor(Long doctorId, Pageable pageable);

    /** Attaches an AI-generated plain-language summary; called from the AI module. */
    LabReportResponse attachAiSummary(Long id, String aiSummary);
}

File: frontend/src/services/labReportService.js
import apiClient from './apiClient';

const labReportService = {
  async request(payload) {
    const { data } = await apiClient.post('/lab-reports', payload);
    return data.data;
  },
  async update(id, payload) {
    const { data } = await apiClient.patch(`/lab-reports/${id}`, payload);
    return data.data;
  },
  async getById(id) {
    const { data } = await apiClient.get(`/lab-reports/${id}`);
    return data.data;
  },
  async getByPatient(patientId, params = {}) {
    const { data } = await apiClient.get(`/lab-reports/patient/${patientId}`, { params });
    return data.data;
  },
  async getByDoctor(doctorId, params = {}) {
    const { data } = await apiClient.get(`/lab-reports/doctor/${doctorId}`, { params });
    return data.data;
  },
};

export default labReportService;
